sqlgen/db_generator.go

```diff
@@ -84,6 +84,9 @@ func GenNewIndex(id int, tbl *Table, w *Weight) *Index {
 		}
 		totalCols[0], totalCols[chosenIdx] = totalCols[chosenIdx], totalCols[0]
 		totalCols = totalCols[1:]
+		if idx.Tp == IndexTypePrimary && chosenCol.defaultVal == "null" {
+			continue
+		}
```

Comment:
It is possible that all the columns are nullable, and the `GenNewIndex()` will panic.

Reply:
right, then the table cannot have partitions and cannot have a primary key. I have changed the `colDef` to make sure the table has at least one not nullable column.